torch/onnx/_internal/exporter/_building.py

```diff
@@ -458,11 +458,29 @@ def _process_python_sequences(
     return named_inputs
 
 
+def _determine_output_number(
+    signature: _schemas.OpSignature, named_attrs: Mapping[str, ValidAttributeType]
+) -> int:
+    """Determine the number of outputs for the node with heuristics."""
+    if signature.domain == "":
+        if signature.name == "BatchNormalization":
+            if not named_attrs.get("training_mode", 0):
+                return 1
+        if signature.name == "Split":
+            num_outputs = named_attrs.get("num_outputs")
+            if num_outputs is not None and isinstance(num_outputs, int):
+                return num_outputs
+            else:
+                logger.warning("Could not determine the number of outputs for Split.")
```

Comment:
Why not just failing? signature.outputs returns 1 or something else in that case?

Reply:
Changed to raise a ValueError